### PR TITLE
Adding SQL server Auditing templates

### PR DESCRIPTION
**Pull request by lubalibu**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: 201-sql-auditing-server-policy-to-blob-storage/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Deploy an Azure SQL Server with Auditing enabled to qrite audit logs to Azure blob storage account
```
Comment (guywald1) [inline 201-sql-auditing-server-policy-to-blob-storage/README.md]:
```suggestion
# Deploy an Azure SQL Server with Auditing enabled to write audit logs to Azure blob storage account
```

File: 201-sql-auditing-server-policy-to-blob-storage/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    },
+    "resources": [{
+      "apiVersion": "2017-03-01-preview",
+      "type": "auditingSettings",
+      "name": "DefaultAuditingSettings",
+      "dependsOn": ["[parameters('sqlServerName')]",
+      "[parameters('storageAccountName')]"],
+      "properties": {
+        "state": "Enabled",
+        "storageEndpoint": "[concat('https://',parameters('storageAccountName'),'.blob.core.windows.net')]",
```
Comment (bmoore-msft) [inline 201-sql-auditing-server-policy-to-blob-storage/azuredeploy.json]: See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#referencing-resource-properties) for how to handle storage account uris

File: 201-sql-auditing-server-policy-to-blob-storage/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        "storageAccountAccessKey": "[listKeys(resourceId('Microsoft.Storage/storageAccounts', parameters('storageAccountName')), providers('Microsoft.Storage', 'storageAccounts').apiVersions[0]).keys[0].value]",
```
Comment (bmoore-msft) [inline 201-sql-auditing-server-policy-to-blob-storage/azuredeploy.json]: remove providers() call and use a literal apiVersion (providers is not deterministic)

File: 201-sql-auditing-server-policy-to-eventhub/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        "description": "The administrator password of the SQL Server."
+      }
+    },
+    "eventHubName": {
+      "type": "string",
+      "metadata": {
+        "description": "Event Hub name."
+      }
+    },
+    "eventHubNamespaceName": {
```
Comment (bmoore-msft) [inline 201-sql-auditing-server-policy-to-eventhub/azuredeploy.json]: generate a unique defaultValue here

File: 201-sql-auditing-server-policy-to-eventhub/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        "description": "The administrator username of the SQL Server."
+      }
+    },
+    "sqlAdministratorLoginPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The administrator password of the SQL Server."
+      }
+    },
+    "eventHubName": {
```
Comment (bmoore-msft) [inline 201-sql-auditing-server-policy-to-eventhub/azuredeploy.json]: add a defaultValue

File: 201-sql-auditing-server-policy-to-eventhub/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+      "defaultValue": "RootManageSharedAccessKey",
+      "metadata": {
+        "description": "Name of Namespace Authorization Rule."
+      }
+    }
+  },
+  "variables": {
+   "diagnosticSettingsName": "SQLSecurityAuditEvents_3d229c42-c7e7-4c97-9a99-ec0d0d8b86c1"
+  },
+  "resources": [{
```
Comment (bmoore-msft) [inline 201-sql-auditing-server-policy-to-eventhub/azuredeploy.json]: format the JSON (e.g. SHIFT+ALT+F in VS Code)

File: 201-sql-auditing-server-policy-to-blob-storage/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+      "defaultValue": "[concat('sqlaudit',uniqueString(resourceGroup().id))]",
+      "metadata": {
+        "description": "The name of the auditing storage account."
+      }
+    }
+  },
+  "variables": {
+    
+  },
+  "resources": [{
```
Comment (bmoore-msft) [inline 201-sql-auditing-server-policy-to-blob-storage/azuredeploy.json]: format the JSON (e.g. SHIFT+ALT+F in VS Code)

File: 201-sql-auditing-server-policy-to-eventhub/azuredeploy.parameters.json [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "sqlAdministratorLogin": {
+      "value": "GEN-UNIQUE"
+    },
+    "sqlAdministratorLoginPassword": {
+      "value": "GEN-PASSWORD"
+    },
+    "eventHubName": {
```
Comment (bmoore-msft) [inline 201-sql-auditing-server-policy-to-eventhub/azuredeploy.parameters.json]: remove these once you have defaultValues in the template

File: 201-sql-auditing-server-policy-to-oms/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+      ],
+      "metadata": {
+        "description": "Select the SKU for OMS workspace"
+      }
+    }
+  },
+  "variables": {
+    "diagnosticSettingsName": "SQLSecurityAuditEvents_3d229c42-c7e7-4c97-9a99-ec0d0d8b86c1"
+  },
+  "resources": [{
```
Comment (bmoore-msft) [inline 201-sql-auditing-server-policy-to-oms/azuredeploy.json]: format the JSON

File: 201-sql-auditing-server-policy-to-oms/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+    "name": "[parameters('omsWorkspaceName')]",
+    "type": "Microsoft.OperationalInsights/workspaces",
+    "apiVersion": "2015-11-01-preview",
```
Comment (bmoore-msft) [inline 201-sql-auditing-server-policy-to-oms/azuredeploy.json]: update apiVersion

File: 201-sql-auditing-server-policy-to-eventhub/azuredeploy.json (hunk 5 above)
Comment (lubalibu) [reply 201-sql-auditing-server-policy-to-eventhub/azuredeploy.json]: done

File: 201-sql-auditing-server-policy-to-eventhub/azuredeploy.json (hunk 4 above)
Comment (lubalibu) [reply 201-sql-auditing-server-policy-to-eventhub/azuredeploy.json]: done

File: 201-sql-auditing-server-policy-to-eventhub/azuredeploy.json (hunk 5 above)
Comment (lubalibu) [reply 201-sql-auditing-server-policy-to-eventhub/azuredeploy.json]: done

File: 201-sql-auditing-server-policy-to-eventhub/azuredeploy.json (hunk 6 above)
Comment (lubalibu) [reply 201-sql-auditing-server-policy-to-eventhub/azuredeploy.json]: done

File: 201-sql-auditing-server-policy-to-oms/azuredeploy.json (hunk 9 above)
Comment (lubalibu) [reply 201-sql-auditing-server-policy-to-oms/azuredeploy.json]: done

File: 201-sql-auditing-server-policy-to-eventhub/azuredeploy.parameters.json (hunk 8 above)
Comment (lubalibu) [reply 201-sql-auditing-server-policy-to-eventhub/azuredeploy.parameters.json]: deone

File: 201-sql-auditing-server-policy-to-oms/azuredeploy.json (hunk 10 above)
Comment (lubalibu) [reply 201-sql-auditing-server-policy-to-oms/azuredeploy.json]: done

File: 201-sql-auditing-server-policy-to-blob-storage/azuredeploy.json (hunk 7 above)
Comment (lubalibu) [reply 201-sql-auditing-server-policy-to-blob-storage/azuredeploy.json]: done

File: 201-sql-auditing-server-policy-to-blob-storage/azuredeploy.json (hunk 3 above)
Comment (lubalibu) [reply 201-sql-auditing-server-policy-to-blob-storage/azuredeploy.json]: done

**Pull request merged**